mysite/blog/views.py

```diff
@@ -23,7 +40,28 @@ def post_detail(request, year, month, day, post):
         publish__month=month,
         publish__day=day,
     )
-    return render(request, "blog/post/detail.html", {"post": post})
+    comments = getCommentsoFPost(post)
+    new_comment = None
+
+    if request.method == "POST":
+        comment_form = CommentForm(data=request.POST)
+        new_comment = check_comment_form_validity_and_save_comment(
+            comment_form, post
+        )
+
+    else:
+        comment_form = CommentForm()
```

Comment:
This view has more than one responsibility. The comment creation should move to posts/<id>/comment/ view

Reply:
added a  new  method for comment view (discussed with kaviya )[https://github.com/Karthikeyantestpress/Django_Blog_Application/pull/10/files#diff-ae5986e281ef46bd77930a4985d5c376ce41689b3f9b85dcae0e6b60c77eecc8:~:text=post/list.html%22-,def%20add_comment(request%2C%20post_id)%3A,),-def%20post_detail(](url)